__tests__/app.test.js

```diff
@@ -339,4 +339,27 @@ describe("/api/users", () => {
         });
       });
   });
-});
\ No newline at end of file
+});
+
+describe.only("/api/articles?topic", () => {
+  test('GET:200 accepts a topic query and responds with an array of articles filtered by topic', () => {
+    return request(app)
+    .get("/api/articles?topic=cats")
+    .expect(200)
+    .then((response) => {
+      const articles = response.body.articles
+      expect(articles.length).toBe(1)
+      articles.forEach((article) => {
+        expect(article.topic).toBe('cats')
+      })
+    })
+  });
+  test('GET:400 responds with status and error message when query topic does not exist', () => {
+    return request(app)
+    .get('/api/articles?topic=food')
+    .expect(400)
+    .then((response) => {
+      expect(response.body.msg).toBe('Bad request')
+    })
+  });
```

Comment:
The 400 test described for topic=food should be a 404

One missing test also:
We have a valid topic which has no associated articles (topic=paper)
This is a 200 and should return an empty array